### PR TITLE
bugfix(otgfs): Fixes a couple issue with OTGFS

### PR DESCRIPTION
**Pull request by Codetector1374**
This change address mainly 2 concerns regarding the OTGFS driver

1. There seems to be potential edge cases where the various transfer polling function is not actually checking for the correct endpoint number before clearing the transfer flag, causing potential lost of transaction.
2. Endpoint is somehow enabled by default, which cause any transaction sent to endpoint prior to actually calling the `endpoint_set_enable` function be lost. This is caused by the hardware behaving differently than what the manual describes.

Beside the two bugfix, it also lays the groundwork for USBHS by extracting some of the common code into the `usb` module.

### CODE REVIEW

File: src/otg_fs/endpoint.rs [hunk 1]
```diff
@@ -469,8 +383,8 @@ where
 
                 let poll_res = if regs.int_fg().read().transfer() {
                     let status = regs.int_st().read();
-                    let res = if status.mask_uis_endp() == 0 {
-                        match status.mask_token() {
+                    if status.mask_uis_endp() == 0 {
+                        let res = match status.mask_token() {
                             UsbToken::OUT => {
                                 if regs.rx_len().read().0 != 0 {
                                     error!("Expected 0 len OUT stage, found non-zero len, aborting");
```
Comment (Dummyc0m) [inline src/otg_fs/endpoint.rs:390]: v.set_r_tog(true) a few lines down is not necessary.

File: src/otg_fs/mod.rs [hunk 2]
```diff
@@ -285,18 +272,31 @@ impl<'d, T: Instance> Bus<'d, T> {
 
         // Mark all other EPs as NAK
         for i in 1..=7 {
+            use embassy_usb_driver::Bus;
             regs.uep_rx_ctrl(i).write(|v| v.set_mask_r_res(EpRxResponse::NAK));
             regs.uep_tx_ctrl(i).write(|v| v.set_mask_t_res(EpTxResponse::NAK));
+            self.endpoint_set_enabled(EndpointAddress::from_parts(i, Direction::In), false);
```
Comment (Dummyc0m) [inline src/otg_fs/mod.rs:278]: add comment

**Review by Dummyc0m — APPROVED** [review]
LGTM

feel free to merge after addressing the comments.

**Review by andelf — APPROVED** [review]
LGTM

**Pull request merged**